test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ServiceBusEndToEndTests.cs

```diff
@@ -449,6 +473,26 @@ public class ServiceBusTestJobs : ServiceBusTestJobsBase
             {
                 output = input;
             }
+
+            [NoAutomaticTrigger]
+            public static async Task ServiceBusBinderTest(
+                string message,
+                int numMessages,
+                Binder binder)
+            {
+                var attribute = new ServiceBusAttribute(BinderQueueName)
+                {
+                    EntityType = EntityType.Queue
+                };
+                var collector = await binder.BindAsync<IAsyncCollector<string>>(attribute);
```

Comment:
I verified that previously each call to BindAsync was resulting in a new MessagingFactory, and with these changes the factory for this connection is created only once